src/components/Timelog/TimeEntryForm/TimeEntryForm.jsx

```diff
@@ -200,9 +200,15 @@ const TimeEntryForm = props => {
       result.dateOfWork = 'Date is required';
     } else {
       const date = moment(inputs.dateOfWork);
+      const today = moment(moment().tz('America/Los_Angeles').format('YYYY-MM-DD'));
       if (!date.isValid()) {
         result.dateOfWork = 'Invalid date';
       }
+      // Admin/Owner can add time entries for any dates. Other roles cannot. 
+      // Editing details of past date is possible for any role.
+      else if (currentUserRole !== 'Owner' && !edit && today.diff(date, 'days') != 0) {
```

Comment:
In the code “Administrator” role is mentioned as owner. So I have checked only owner. Please correct me if I am wrong. Thanks.